### ISSUE TITLE
v9 regression with useSpring set function + useLayoutEffect

### ISSUE THREAD

**Issue by brunolemos** [react-spring/react-spring#768]
## 🐛 Bug Report

I'm using the `const [springAnimatedStyles, setSpringAnimatedStyles] = useSpring(() => ({ color: 'xxx' }))` syntax.

The `springAnimatedStyles` variable contains the animated colors for the current app theme. I share this same animated value with multiple components all over the app. 
 
When changing the theme, I call `setSpringAnimatedStyles`. On v8 this worked correctly, all components using `springAnimatedStyles` would show the new theme colors.

On v9 this stopped happening, now it keeps showing the old theme color. (see screenshot; the sidebar and the header are using `springAnimatedStyles` so they didn't get updated. the rest of the app is using css variables so it got the new colors)

~I think the regression is related sharing animated value between multiple components but not sure. Might be related to #745 based on the title.~

![AwesomeScreenshot-localhost--2019-07-30_7_09](https://user-images.githubusercontent.com/619186/62169750-7fad9e00-b2ff-11e9-8dc4-07284f33dba9.png)

## To Reproduce

Call `setSpringAnimatedStyles` inside a `useLayoutEffect` (see codesandbox below)

## Expected behavior

![image](https://user-images.githubusercontent.com/619186/62170005-46c1f900-b300-11e9-96e4-ceb231f8a747.png)

## Link to repro (highly encouraged)

https://codesandbox.io/embed/hungry-williams-lvm5t

## Environment

- `react-spring` 9.0.0-beta.31
- `react` v16.8.6
- `react-native-web` v0.11.5

**Comment by brunolemos** [comment]
Found a workaround with these two changes:

- `useLayoutEffect` -> `useEffect`
- Wrapping the `set` call with a `setTimeout` of `1`

Maybe it's some kind of race condition?

> PS: This solution is not ideal because now there is a small delay on the theme change of these specific components.

**Comment by brunolemos** [comment]
For some reason the `useLayoutEffect` -> `useEffect` change only improves things on development, on production the bug still occurs 🤔

![image](https://user-images.githubusercontent.com/619186/62184009-b2bd5500-b332-11e9-887e-d53a5b37512e.png)

**Comment by brunolemos** [comment]
Tried to make a codesandbox using react-native-web but as soon as I import `react-spring/native` I get an error:

```
ModuleNotFoundError
Could not find module in path: '@react-spring/native' relative to '/node_modules/react-spring/native.js'
```

**Comment by brunolemos** [comment]
@aleclarson I was able to make a small reproducible demo! https://codesandbox.io/embed/hungry-williams-lvm5t

I see the bug was introduced on v9 beta 25, 26 or 27. It works perfectly on v8 and on v9 beta 24.
This should help the investigation.

What do you think is causing this?

**Comment by brunolemos** [comment]
Bad news is that the codesandbox works on beta 24 100% of the times, but on my app it is still broken.

But who knows, maybe if the bug gets fixed for the codesandbox reproduction that it will fix on my app as well.

**Comment by aleclarson** [comment]
This is a bug, thanks for reporting! ❤️ 

Currently, you can't call the `useSpring/s` setter inside `useLayoutEffect`, because the setter doesn't have its `Controller` references updated until the `useEffect` phase ([see here](https://github.com/react-spring/react-spring/blob/d6a45662c9a55f6a8fe7af4c1fec4a76afe6b563/packages/core/src/useSprings.js#L66-L85)).

There *might* be other bugs here, too. We'll find out when I get around to fixing the `useLayoutEffect` bug (not sure when that'll be).

**Comment by i-am-the-slime** [comment]
Is this still true?

**Comment by aleclarson** [comment]
Fixed in `9.0.0-rc.2` (#985)
